7a_temp_coop_munge/src/data_parsers/parse_Iowa_DNR_LimnoProfiles_2000_2020.R

```diff
@@ -0,0 +1,507 @@
+                                          htm_verbose = F)) %>%
+    dplyr::mutate(date = as.Date(sampling_date_time),
+                  time = format(sampling_date_time, '%H:%M'),
+                  Timezone = NA) %>%
+    dplyr::select(date, time, Timezone, depth, temp, lakeid)
+
+  # Clean data from 2017a and 2017b----------------------------------------------
+  #' 2017a data does not have date or time
+  #' 2017b data does have date and time. Lake name needs to be parsed out of file.
+
+  dat_2017b <- files_2017b %>%
+    purrr::map_dfr(~ parse_2017_2020_data(.,
+                                          keep_cols = c('DATE', 'TIME', 'SITE',
+                                                        'Temp...F.', 'Depth..m.'),
+                                          lakeid_col = 3,
+                                          temp_as_f = T,
+                                          use_readr = F)) %>%
+    dplyr::mutate(Timezone = NA) %>%
+    dplyr::rename(depth = depth..m.,
+           temp = temp...f.) %>%
+    dplyr::select(date, time, Timezone, depth, temp, lakeid)
+
+  # Clean data from 2018-2020 ---------------------------------------------------
+  ## `2019_R3_Profiles.csv` is in an unexpected format. it was faster to just munge
+  ## separately
+
+  dat_2018_2020_1 <- files_2018_2020[c(1:5, 7:9)] %>%
+    purrr::map_dfr(~ parse_2017_2020_data(.,
+                                          keep_cols = c('Lake', 'Date', 'Time',
+                                                        'Depth (m)', 'Temperature (Deg C)'),
+                                          lakeid_col = 1,
+                                          temp_as_f = F,
+                                          use_readr = T)) %>%
+    dplyr::rename(depth = `depth (m)`,
+           temp = `temperature (deg c)`)
+
+
+  dat_2018_2020_2 <- files_2018_2020[6] %>%
+    purrr::map_dfr(~ parse_2017_2020_data(.,
+                                          keep_cols = c('DATE', 'DATE.1', 'TIME',
+                                                        'Depth..m.', 'Temperature..Deg.C.'),
+                                          lakeid_col = 1,
+                                          temp_as_f = F,
+                                          use_readr = F)) %>%
+    dplyr::rename(depth = depth..m.,
+           temp = temperature..deg.c.)
+
+  # clean data -------------------
+  data_clean <- dplyr::bind_rows(dat_2000_2003,
+                          dat_2004_2007,
+                          dat_2009,
+                          dat_2010,
+                          dat_2011_2016,
+                          dat_2017b,
+                          dat_2018_2020_1,
+                          dat_2018_2020_2) %>%
+    dplyr::mutate(Iowa_ID = paste('Iowa_', lakeid, sep = ''), # for matching the xwalk
+      DateTime = as.Date(date)) %>% # there is a mix of POSIXct and Date in the indiv dfs
+    dplyr::select(! c(lake, lakeid, date)) #holdover column from 2017-2020 data
```

Comment:
If it isn't too wordy, would you consider `select` where you are naming the columns you are keeping instead of the ones you are removing? That is a bit easier to read and understand what the final output is w/o digging into the code. If the columns are too numerous though, it doesn't make sense.

Reply:
Absolutely